File: src/typescript/transaction-builder/finalization/proto-form-signature-build.ts
import { createHiveChain } from '@hiveio/wax';

// Initialize hive chain interface
const chain = await createHiveChain();

// Initialize a transaction object
const tx = await chain.createTransaction();

// Declare example operation
const operation = {
  vote: {
    voter: "voter",
    author: "author",
    permlink: "test-permlink",
    weight: 2200
  }
};

// Push operation into the transction
tx.pushOperation(operation);

// Supplement a transaction with an externally generated signature.
tx.sign('signature...');

console.log(tx.toApi());

File: src/python/transaction/finalization/proto_form_signature_build.py
from __future__ import annotations

import asyncio

from wax import create_hive_chain
from wax.proto.operations import vote

# Initialize hive chain interface
chain = create_hive_chain()

WALLET_NAME = "wallet_name"
WALLET_PASSWORD = "wallet_password"
key = chain.get_private_key_from_password("account_name", role="active", password="password")
public_key = key.associated_public_key
private_key = key.wif_private_key


async def main():
    # Initialize a transaction object
    tx = await chain.create_transaction()

    # Declare example operation
    operation = vote(voter="voter", author="test-author", permlink="test-permlink", weight=2200)

    # Push operation into the transaction
    tx.push_operation(operation)

    # Supplement a transaction with an externally generated signature.
    tx.add_signature("deadc0de")
    print(tx.to_api())


asyncio.run(main())
